common/src/main/java/org/jboss/eap/qe/ts/common/docker/Docker.java

```diff
@@ -0,0 +1,188 @@
+
+        cmd.add("docker");
+        cmd.add("run");
+        cmd.add("--name");
+        cmd.add(uuid);
+
+        for (String port : ports) {
+            cmd.add("-p");
+            cmd.add(port);
+        }
+
+        for (Map.Entry<String, String> envVar : environmentVariables.entrySet()) {
+            cmd.add("-e");
+            cmd.add(envVar.getKey() + "=" + envVar.getValue());
+        }
+
+        cmd.add(image);
+
+        cmd.addAll(commandArguments);
+
+        System.out.println(Ansi.ansi().reset().a("Starting container ").fgCyan().a(name).reset()
+                .a(" with ID ").fgYellow().a(uuid).reset());
+
+        Process dockerRunProcess = new ProcessBuilder()
+                .redirectErrorStream(true)
+                .command(cmd)
+                .start();
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        outputPrinter.execute(() -> {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(dockerRunProcess.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    System.out.println(Ansi.ansi().fgCyan().a(name).reset().a("> ").a(line));
+
+                    if (awaitedLogLine != null && line.contains(awaitedLogLine)) {
+                        latch.countDown();
+                    }
+                }
+            } catch (IOException ignored) {
+            }
+        });
+
+        if (awaitedLogLine != null) {
+            if (latch.await(waitTimeoutInMillis, TimeUnit.MILLISECONDS)) {
+                return;
+            }
+
+            try {
+                stop();
+            } catch (Exception ignored) {
+                // ignored because we're about to throw another exception
+            }
+
+            throw new TimeoutException("Container '" + name + " (" + uuid + ")' didn't print '"
+                    + awaitedLogLine + "' in " + waitTimeoutInMillis + " ms");
+        }
+    }
+
+    private void stop() throws IOException, InterruptedException {
+        System.out.println(Ansi.ansi().reset().a("Stopping container ").fgCyan().a(name).reset()
+                .a(" with ID ").fgYellow().a(uuid).reset());
+
+        new ProcessBuilder()
+                .command("docker", "stop", uuid)
+                .start()
+                .waitFor(10, TimeUnit.SECONDS);
+
+        outputPrinter.shutdown();
+        outputPrinter.awaitTermination(10, TimeUnit.SECONDS);
+
+        new ProcessBuilder()
+                .command("docker", "rm", uuid)
+                .start()
+                .waitFor(10, TimeUnit.SECONDS);
+    }
+
+    // ---
+
+    @Override
+    protected void before() throws Throwable {
+        start();
+    }
+
+    @Override
+    protected void after() {
+        try {
+            stop();
+        } catch (IOException | InterruptedException e) {
+            System.out.println(Ansi.ansi().reset().a("Failed stopping container ").fgCyan().a(name).reset()
```

Comment:
What about Java logging or Slf4j? Also, do we want to swallow the exception and continue if container is not stopped?